Fix gps rescue throttle pid gains

ctzsnooze opened the pull request:
This PR is a simple one to:
- update the GPS Rescue Altitude control Debug traces (PID and altitude),
- add GPS Rescue Velocity control Debug traces (PID and velocity), and
- update the RTH graphs and labels.

It is intended to properly display the new GPS Rescue debug traces from [11579](https://github.com/betaflight/betaflight/pull/11579)

blckmn commented:
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

haslinghuis commented on js/graph_config.js:
```diff
@@ -736,6 +736,105 @@ GraphConfig.load = function(config) {
+                                    inputRange: 500,
+                                    outputRange: 1.0,
+                                };
+                            case 'debug[2]': // Throttle D
```
Same

ctzsnooze replied:
Hopefully OK now

haslinghuis commented on js/graph_config.js:
```diff
@@ -736,6 +736,105 @@ GraphConfig.load = function(config) {
+                                    inputRange: 500,
+                                    outputRange: 1.0,
+                                };
+                            case 'debug[1]': // Throttle I
```
Best to move this line to 775 (and remove the inner block) as fall through case as the inner block is repeated.

ctzsnooze replied:
OK now I hope

haslinghuis commented on js/graph_config.js:
```diff
@@ -736,6 +736,105 @@ GraphConfig.load = function(config) {
+                                    inputRange: 500,
+                                    outputRange: 1.0,
+                                };
+                            case 'debug[2]': // Throttle D
```
Same.

haslinghuis commented on js/graph_config.js:
```diff
@@ -736,6 +736,105 @@ GraphConfig.load = function(config) {
+                                    inputRange: 500,
+                                    outputRange: 1.0,
+                                };
+                            case 'debug[3]': // Velocity cm/s
```
Same.

haslinghuis reviewed (changes requested):
Please remove duplicate blocks (to remove the code duplication errors).

McGiverGim commented on js/flightlog_fields_presenter.js:
```diff
@@ -1056,7 +1081,15 @@ function FlightLogFieldPresenter() {
                     }
                 case 'GPS_RESCUE_THROTTLE_PID':
-                        return value.toFixed(0);
+                    switch (fieldName) {
+                        // debug 0 = Throttle P
+                        // debug 1 = Throttle I
+                        // debug 3 = Throttle D
+                        case 'debug[3]': // current altitude in m
+                            return (value / 100).toFixed(2) + 'm';
+                        default:
+                            return value.toFixed(0);
+                    }
```
I suppose that `debug 3` is wrong and is `debug 2`. Move the comments to the default to let clear where it will enter. For example:
```suggestion
                    switch (fieldName) {
                        case 'debug[3]': // current altitude in m
                            return (value / 100).toFixed(2) + 'm';                            

                        // debug 0 = Throttle P
                        // debug 1 = Throttle I
                        // debug 2 = Throttle D                            
                        default:
                            return value.toFixed(0);
                    }
```

ctzsnooze replied:
Thanks, have done as suggested.

McGiverGim commented on js/flightlog_fields_presenter.js:
```diff
@@ -1125,6 +1158,32 @@ function FlightLogFieldPresenter() {
                             return value.toFixed(0) + 'us';
                     }
+                case 'GPS_RESCUE_VELOCITY':
+                    switch (fieldName) {
+                        // debug 0 = Velocity P
+                        // debug 1 = Velocity I
+                        // debug 2 = Velocity D
+                        case 'debug[3]': // Velocity cm/s
+                            return (value / 100).toFixed(2) + 'm/s';
+                        default:
+                            return value.toFixed(0);
+                    }
```
Same than before:
```suggestion
                    switch (fieldName) {
                        case 'debug[3]': // Velocity cm/s
                            return (value / 100).toFixed(2) + 'm/s';

                        // debug 0 = Velocity P
                        // debug 1 = Velocity I
                        // debug 2 = Velocity D
                        default:
                            return value.toFixed(0);
                    }
```

ctzsnooze replied:
OK thanks

McGiverGim commented on js/flightlog_fields_presenter.js:
```diff
@@ -1125,6 +1158,32 @@ function FlightLogFieldPresenter() {
+                        default:
+                            return value.toFixed(0);
+                    }
+                case 'GPS_RESCUE_HEADING':
+                    switch (fieldName) {
+                        // debug 0 = error angle tenths of a degree
+                        case 'debug[0]': // error angle tenths of a degree
+                            return (value / 10).toFixed(1) + 'deg';
+                        // debug 1 = Velocity I
+                        case 'debug[1]': // angle to home in tenths of a degree
+                            return (value / 10).toFixed(1) + 'deg';
+                        // debug 2 = Velocity D
+                        case 'debug[2]': // commanded yaw rate in tenths of a degree
+                            return (value / 10).toFixed(1) + 'deg/s';
+                        case 'debug[3]': // roll angle in hundredths of a degree
+                            return (value / 100).toFixed(2) + 'deg';
+                        default:
+                            return value.toFixed(0);
+                    }
```
The same, group cases, and some comments seems wrong from a copy&paste.
```suggestion
                case 'GPS_RESCUE_HEADING':
                    switch (fieldName) {
                        case 'debug[0]': // error angle tenths of a degree
                        case 'debug[1]': // angle to home in tenths of a degree                        
                            return (value / 10).toFixed(1) + 'deg';
                        case 'debug[2]': // commanded yaw rate in tenths of a degree
                            return (value / 10).toFixed(1) + 'deg/s';
                        case 'debug[3]': // roll angle in hundredths of a degree
                            return (value / 100).toFixed(2) + 'deg';
                        default:
                            return value.toFixed(0);
                    }
```

ctzsnooze replied:
thanks, updated

McGiverGim reviewed (changes requested):
My comments go in the same line than @haslinghuis. Try to group `case` elements, in this way is clear that various elements have exactly the same behaviour. For the rest the PR is ok.

ctzsnooze commented:
Not sure if I addressed all the issues above, if not please let me know.

Added another debug for https://github.com/betaflight/betaflight/pull/11579, tested.

McGiverGim commented:
> Not sure if I addressed all the issues above, if not please let me know.

I think my review has not been fixed, I don't see any change in the code about it.

McGiverGim commented on js/flightlog_fields_presenter.js:
```diff
@@ -1056,7 +1088,16 @@ function FlightLogFieldPresenter() {
                     }
                 case 'GPS_RESCUE_THROTTLE_PID':
-                        return value.toFixed(0);
+                    switch (fieldName) {
+                        case 'debug[0]': // Throttle P added uS
+                        case 'debug[1]': // Throttle D added * uS
+                            return value.toFixed(0) + 'uS';
```
the `uS` unit here is different from line 1031 or 1035, where is lowercase. I think all of them must be equal.
```suggestion
                    switch (fieldName) {
                        case 'debug[0]': // Throttle P added us
                        case 'debug[1]': // Throttle D added * us
                            return value.toFixed(0) + 'us';
```

ctzsnooze replied:
thanks, done

McGiverGim commented on js/flightlog_fields_presenter.js:
```diff
@@ -999,8 +1027,12 @@ function FlightLogFieldPresenter() {
                     }
                 case 'RTH':
                     switch(fieldName) {
+                        case 'debug[0]':
+                            return value.toFixed(0) + 'us';
                         case 'debug[1]':
                             return (value / 100).toFixed(1) + 'deg';
+                        case 'debug[2]':
+                            return value.toFixed(0) + 'us';
                         default:
                             return value.toFixed(0);
                     }
```
Is better to have them in different order to not duplicate code:
```suggestion
                    switch(fieldName) {
                        case 'debug[0]':
                        case 'debug[2]':                        
                            return value.toFixed(0) + 'us';
                        case 'debug[1]':
                            return (value / 100).toFixed(1) + 'deg';
                        default:
                            return value.toFixed(0);
                    }
```

ctzsnooze replied:
Have revised these and hopefully is OK now.

McGiverGim commented on js/flightlog_fielddefs.js:
```diff
@@ -339,6 +339,10 @@ var
+            "DEBUG_RX_STATE_TIME",
+            "GPS_RESCUE_VELOCITY",
+            "GPS_RESCUE_HEADING",
+            "GPS_RESCUE_TRACKING"
```
Sonar issue: missing trailing comma

ctzsnooze replied:
thanks, fixed

McGiverGim commented on js/flightlog_fields_presenter.js:
```diff
@@ -1125,6 +1166,36 @@ function FlightLogFieldPresenter() {
+                    }
+                case 'GPS_RESCUE_HEADING':
+                    switch (fieldName) {
+                        case 'debug[0]': // Rescue yaw rate deg/s * 10 up to +/- 90 
```
Sonar issue: remove the trailing space at the end

ctzsnooze replied:
thanks

McGiverGim reviewed (changes requested):
Apart from my other review, Sonar is complaining about some things, I've marked the basic. The others, I think we can ignore them.

ctzsnooze commented on js/graph_config.js:
```diff
@@ -736,6 +736,105 @@ GraphConfig.load = function(config) {
+                                    inputRange: 500,
+                                    outputRange: 1.0,
+                                };
+                            case 'debug[1]': // Throttle I
```
I tried to fix this, hopefully it is OK now

McGiverGim commented on js/flightlog_fields_presenter.js:
```diff
@@ -999,10 +1027,14 @@ function FlightLogFieldPresenter() {
                     switch(fieldName) {
+                        case 'debug[0]':
+                            return value.toFixed(0) + 'us'; // rescue throttle 1000-2000
                         case 'debug[1]':
-                            return (value / 100).toFixed(1) + 'deg';
+                            return (value / 100).toFixed(1) + 'deg'; // rescue pitch angle
+                        case 'debug[2]':
+                            return value.toFixed(0) + 'us'; // altitude adjustment min to max throttle
                         default:
-                            return value.toFixed(0);
+                            return value.toFixed(0); // coded values to 2000
```
I think this is the only one missing. The debug[0] and debug[2] are the same, so group this two and I think we are ready.

ctzsnooze replied:
Thank you.  I hope it is fixed now.
Should I squash?

McGiverGim reviewed (changes requested):
The latest one, please ;)

McGiverGim commented on js/graph_config.js:
```diff
@@ -736,6 +736,129 @@ GraphConfig.load = function(config) {
+                            }
+                    case 'GPS_RESCUE_HEADING':
+                        switch (fieldName) {
+                            case 'debug[0]': // Rescue yaw rate deg/s * 10 up to +/- 90 
```
Sonar issue: trailing space at the end

McGiverGim reviewed (commented):
I think is ok now. A missing trailing space, but for the rest is ok to me. If you squash the commits, please remove the trailing space too ;)

ctzsnooze commented:
Closing, replaced with #593
